lunes_cms/cms/migrations/0007_document_created_by_link.py

```diff
@@ -0,0 +1,55 @@
+# Generated by Django 3.2.13 on 2022-07-05 11:32
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+# pylint: disable=unused-argument
+def link_created_by(apps, schema_editor):
+    """
+    Convert the ``created_by`` char field to a foreign key
+
+    :param apps: The configuration of installed applications
+    :type apps: ~django.apps.registry.Apps
+
+    :param schema_editor: The database abstraction layer that creates actual SQL code
+    :type schema_editor: ~django.db.backends.base.schema.BaseDatabaseSchemaEditor
+    """
+    Document = apps.get_model("cms", "Document")
+    Group = apps.get_model("auth", "Group")
+    for document in Document.objects.filter(created_by__isnull=False):
+        group, created = Group.objects.get_or_create(name=document.created_by)
+        document.created_by_link = group
+        document.save()
+
```

Comment:
```suggestion

# pylint: disable=unused-argument
def reverse_created_by(apps, schema_editor):
    """
    Convert the ``created_by`` field from foreign key to a char field

    :param apps: The configuration of installed applications
    :type apps: ~django.apps.registry.Apps

    :param schema_editor: The database abstraction layer that creates actual SQL code
    :type schema_editor: ~django.db.backends.base.schema.BaseDatabaseSchemaEditor
    """
    Document = apps.get_model("cms", "Document")
    Group = apps.get_model("auth", "Group")
    for document in Document.objects.filter(created_by__isnull=False):
        group, created = Group.objects.get_or_create(id=document.created_by_link.id)
        document.created_by = group.name
        document.save()

```
Since in theory, this migration can also be reversed, we could also define a function to convert the field from a foreign key back to a char field. This can be helpful if e.g. the new code contains a bug and the migration needs to be rolled back on the production system. For more info, see the [docs](https://docs.djangoproject.com/en/4.0/topics/migrations/#reversing-migrations).

Reply:
Didn't you like this suggestion, or did you inadvertently not apply it?

Reply:
I added the suggestion to my local copy but hadn't pushed it yet. Now it should be applied :)